zkevm-circuits/src/witness/tx.rs

```diff
@@ -519,15 +906,190 @@ pub(super) fn tx_convert(
     }
 }
 
-impl From<&Transaction> for SignedTransaction {
-    fn from(tx: &Transaction) -> Self {
-        Self {
-            tx: tx.clone(),
-            signature: Signature {
-                v: tx.v,
-                r: tx.r,
-                s: tx.s,
-            },
+#[cfg(test)]
+mod tests {
+    use crate::witness::{tx::Challenges, RlpTag, Transaction};
+    use eth_types::{address, geth_types::TxTypes, word, Address, ToBigEndian, ToScalar};
+    use ethers_core::{
+        types::{NameOrAddress, Signature, Transaction as EthTransaction, TransactionRequest},
+        utils::rlp::{Decodable, Rlp},
+    };
+    use halo2_proofs::{circuit::Value, dev::unwrap_value, halo2curves::bn256::Fr};
+
+    fn rlc(be_bytes: &[u8], rand: Fr) -> Fr {
+        be_bytes
+            .iter()
+            .fold(Fr::zero(), |acc, &byte| acc * rand + Fr::from(byte as u64))
+    }
+
+    #[test]
+    fn test_rlp_pre_eip155() {
+        // the tx is downloaded from https://etherscan.io/getRawTx?tx=0x5c504ed432cb51138bcf09aa5e8a410dd4a1e204ef84bfed1be16dfba1b22060
+        let raw_tx_rlp_bytes = hex::decode("f86780862d79883d2000825208945df9b87991262f6ba471f09758cde1c0fc1de734827a69801ca088ff6cf0fefd94db46111149ae4bfc179e9b94721fffd821d38d16464b3f71d0a045e0aff800961cfce805daef7016b9b675c137a6a41a548f7b60a3484c06a33a")
+            .expect("decode tx's hex shall not fail");
+
+        // NOTE: the decode method in ethers-rs (0.17.0) is buggy
+        // let eth_tx = EthTransaction::decode(&Rlp::new(&raw_tx_rlp_bytes))
+        //     .expect("decode tx's rlp bytes shall not fail");
+
+        let eth_tx = TransactionRequest::new()
+            .nonce(word!("0x00"))
+            .gas_price(word!("0x2d79883d2000"))
+            .gas(word!("0x5208"))
+            .to(address!("0x5df9b87991262f6ba471f09758cde1c0fc1de734"))
+            .value(word!("0x7a69"));
+
+        let eth_sig = Signature {
+            r: word!("0x88ff6cf0fefd94db46111149ae4bfc179e9b94721fffd821d38d16464b3f71d0"),
+            s: word!("0x45e0aff800961cfce805daef7016b9b675c137a6a41a548f7b60a3484c06a33a"),
+            v: 0x1c,
+        };
+
+        let eth_tx_bytes = eth_tx.rlp_signed(&eth_sig).to_vec();
+        assert_eq!(eth_tx_bytes, raw_tx_rlp_bytes);
+
+        // testing sign RLP decoding
+        let tx = Transaction::new_from_rlp_unsigned_bytes(
+            TxTypes::PreEip155,
+            eth_tx.rlp_unsigned().to_vec(),
+        );
+        let evm_word = Fr::from(0x100);
+        let keccak_input = Fr::from(0x10000);
```

Comment:
We should use another mocked evm word for testing. For example, changing this to `0x101` we can find that the unit test will fail.